managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedCursorImpl.java

```diff
@@ -1088,6 +1089,11 @@ public DelayCheckForNewEntriesTask(OpReadEntry op, ReadEntriesCallback callback,
         @Override
         public void run() {
             synchronized (pendingReadOpMutex) {
+                if (cancelled) {
+                    // change state back to init in order to execute cancel()
+                    state = DelayCheckForNewEntriesTaskState.INIT;
+                    return;
+                }
```

Comment:
This seems to indicate that the original state model has some gaps. Could you explain more about your findings about the issue itself? Please also update the description of the PR with some more context.